core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -797,7 +795,14 @@ class ReplicaManager(val config: KafkaConfig,
             brokerTopicStats.topicStats(topicPartition.topic).failedProduceRequestRate.mark()
             brokerTopicStats.allTopicsStats.failedProduceRequestRate.mark()
             error(s"Error processing append operation on partition $topicPartition", t)
-            (topicPartition, LogAppendResult(LogAppendInfo.unknownLogAppendInfoWithLogStartOffset(logStartOffset), Some(t)))
+
```

Comment:
Let's create another `case` for `InvalidRecordException | InvalidTimestampException` before Throwable to make the logic clearer.

Reply:
ack